lib/util.cpp

```diff
@@ -542,3 +542,38 @@ bool process_exists(int pid) {
 }
 
 #endif
+
+unsigned int random_int() {
+    unsigned int n;
+#if defined(_WIN32)
+#if defined(__CYGWIN32__)
+    HMODULE hLib=LoadLibrary((const char *)"ADVAPI32.DLL");
+#else
+    HMODULE hLib=LoadLibraryA("ADVAPI32.DLL");
+#endif
+    if (!hLib) {
+        fprintf(stderr, "Error: can't load ADVAPI32.DLL\n");
+    }
+    BOOLEAN (APIENTRY *pfn)(void*, ULONG) =
+        (BOOLEAN (APIENTRY *)(void*,ULONG))GetProcAddress(hLib,"SystemFunction036");
+    if (pfn) {
+        char buff[32];
+        ULONG ulCbBuff = sizeof(buff);
+        if(pfn(buff,ulCbBuff)) {
+            // use buff full of random goop
+            memcpy(&n,buff,sizeof(n));
+        }
+    }
+    FreeLibrary(hLib);
+#else
+    FILE* f = fopen("/dev/random", "r");
+    if (!f) {
+        fprintf(stderr, "Error: can't open /dev/random\n");
+    }
+    if (1 != fread(&n, sizeof(n), 1, f)) {
+        fprintf(stderr, "Error: can't read from /dev/random\n");
+    }
+    fclose(f);
+#endif
+    return n;
```

Comment:
What will be returned in case of an error?
We should indicate somehow that the function returns a proper result.